### PR TITLE
updated meteo_access functions to include WCC, began draft for gpm access, added citizen sci obs ecoregion/state

### PR DESCRIPTION
**Pull request by nayhur**

### CODE REVIEW

File: R/citsci_preprocess.R [hunk 1]
```diff
@@ -56,12 +56,40 @@ make_tz_table <- function(){
...
   terra::extract(r, locs) %>% as.numeric()
 }
 
+#' Geolocate location to assign state and ecoregion association
+#'
+#' @return State and ecoregion
+#'
+get_geoinfo <- function(lon_obs, lat_obs){
+
+  locs = sf::st_as_sf(data.frame(lon_obs, lat_obs),
```
Comment (SnowHydrology) [inline R/citsci_preprocess.R:75]: @nayhur, please add the `sf` package to the list of `Imports` in the `DESCRIPTION` file in the main level of the repo (https://github.com/SnowHydrology/rainOrSnowTools/blob/e81ad1d3d7913f702d3fbce8a40c3ba793e71895/DESCRIPTION#L13).

File: R/citsci_preprocess.R [hunk 2]
```diff
@@ -56,12 +56,40 @@ make_tz_table <- function(){
...
   locs = cbind(lon_obs, lat_obs)
   r = terra::rast("/vsicurl/https://prd-tnm.s3.amazonaws.com/StagedProducts/Elevation/13/TIFF/USGS_Seamless_DEM_13.vrt")
   terra::extract(r, locs) %>% as.numeric()
 }
 
+#' Geolocate location to assign state and ecoregion association
+#'
+#' @return State and ecoregion
+#'
+get_geoinfo <- function(lon_obs, lat_obs){
```
Comment (SnowHydrology) [inline R/citsci_preprocess.R]: @nayhur, let's split this into 3 functions:

- `get_eco_level3`
- `get_eco_level4`
- `get_state`

Or conversely, you could keep the generic `get_geoinfo` function and have the type of data (ecoregion level vs. state) be an argument to the function. We only want this function to return a single value or column of values at a time.

File: data-raw/misc_metadata.R
Comment (SnowHydrology) [inline data-raw/misc_metadata.R:1]: @nayhur, please delete this file and add a new one called `geographic_metadata.R`. 

We'll want to use the following code from @mikejohnson51 to get ecoregion level 4 data:

```
library(glue)
library(rmapshaper)
library(sf)

################################################################################
# Script for ecoregion and state data
################################################################################

get_ecoregion = function(level = 4){
  
  temp = tempfile()
  eco = tempfile()
  
  URL = glue("[https://gaftp.epa.gov/EPADataCommons/ORD/Ecoregions/us/us_eco_l{level}_state_boundaries.zip](https://gaftp.epa.gov/EPADataCommons/ORD/Ecoregions/us/us_eco_l%7Blevel%7D_state_boundaries.zip)")
  
  download.file(URL, temp)
  
  # Unzip the contents of the temp and save unzipped contents in 'eco'
  unzip(zipfile = temp, exdir = eco)
  
  # Read the shapefile
  sf::read_sf(eco) %>%
    st_make_valid() %>% 
    ms_simplify(keep_shapes = TRUE) %>% 
    st_transform(crs = 4326) 
}

ecoregions_states = get_ecoregion(4)
usethis::use_data(ecoregions_states, overwrite = TRUE)
```

File: R/sysdata.rda
Comment (SnowHydrology) [inline R/sysdata.rda:1]: @nayhur, please update the geographic metadata per my other comment. Once that's done, rerun `meteo_metadata.R` to update `sysdata.rda`.

File: data-raw/misc_metadata.R
Comment (SnowHydrology) [reply data-raw/misc_metadata.R:1]: And once the code is updated, we'll add another R file to the `R` directory that includes this text:

```
#' Ecoregions Levels 2, 3, and 4 plus US State Boundaries

"ecoregions_states"
```

**Review by SnowHydrology — CHANGES_REQUESTED** [review]
@nayhur, thanks for getting this started! Can you address the changes I suggested for the ecoregion and state functions? After those are integrated, I can get started on the met access function for WCC. Thank you!

File: R/sysdata.rda
Comment (mikejohnson51) [reply R/sysdata.rda:1]: If using the above approach and documented dataset, the outputs will go into `data` and `sysdata.rda` should be deleted. If you want to keep `sysdata.rds` then add `intern = TRUE` to `use_data()` and dont add the new data documentation R file. Happy to help if needed!

Comment (SnowHydrology) [reply R/sysdata.rda:1]: @mikejohnson51, we'll continue to use `sysdata.rda` for the met network metadata.

File: data-raw/misc_metadata.R
Comment (nayhur) [reply data-raw/misc_metadata.R:1]: Does it matter what the file name is?

**Pull request closed without merging**